plotly/offline/offline.py

```diff
@@ -325,13 +373,38 @@ def plot(figure_or_data,
             else:
                 plotly_js_script = ''
 
+            if image:
+                warnings.warn('For higher resolution images and more export '
+                              'options, consider making requests to the '
+                              'Plotly image servers. Type help(\'py.image\') '
+                              'for more details.', UserWarning)
+
+                # write the download script:
+                script = ('<script>'
+                          'function downloadimage(format, height, width,'
+                          ' filename) {{'
+                          'var p = document.getElementById(\'{plot_id}\');'
+                          'Plotly.downloadImage(p, {{format: format, height: '
+                          'height, width: width, filename: filename}});'
+                          '}};'
+                          'if(confirm(\'Do you want to save this image as '
+                          '{filename}.{format}?\')) {{'
+                          'downloadimage(\'{format}\', {height}, {width}, '
+                          '\'{filename}\');}}'
+                          '</script>'
+                          ).format(format=format, width=_width, height=_height,
+                                   filename=image_filename, plot_id=plotdivid)
```

Comment:
Is this script the same as above? Can we factor this out to be a little more DRY 🐫.